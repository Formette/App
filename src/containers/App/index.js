// @flow
import React, { PureComponent } from "react";
import { withRouter } from "react-router-dom";
import { compose } from "react-apollo";
//Components
import { Alert } from "../../components/atoms";
import { Footer } from "../../components/organisms";
import Navbar from "./Navigation";
import MobileNavbar from "./MobileNavigation";
//Context
import { UserContext } from "../../context/UserContext";
//hocs
import { withUser } from "../../hocs";
//Styles
import styled, { withTheme } from "styled-components";
//Pages Navigation
import Router from "../../Router";

class App extends PureComponent {
  state = {
    accountConfirmed: true
  };
  componentWillReceiveProps(nextProps) {
    const { confirmed } = nextProps.user.state.profile;
    this.setState({ accountConfirmed: confirmed });
  }
  render() {
    return (
      <div className={this.props.className}>
        <UserContext.Consumer>
          {context => (
            <Navbar brand="Formette Î²" username={context.state.userName} />
          )}
        </UserContext.Consumer>
        {!this.state.accountConfirmed && (
          <Alert className="alert-warning" role="alert">
            You have not activated your account yet.{" "}
            <a href="#/confirm" className="alert-link">
              Click here to activate.
            </a>
          </Alert>
        )}
        <div className="container-fluid content">
          <Router />
        </div>
        <Footer />
        <MobileNavbar />
      </div>
    );
  }
}

const AppWithStyles = styled(App)`
  margin-top: 100px;
  .content {
    min-height: calc(88vh - 70px);
  }
`;

export default compose(
  withUser,
  withRouter,
  withTheme
)(AppWithStyles);
