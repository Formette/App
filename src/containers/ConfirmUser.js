// @flow
import React from "react";
import { graphql, compose } from "react-apollo";
//Components
import AlertContainer from "react-alert";
import AuthLayout from "../components/organisms/AuthLayout";
import { Input, Button, Link, Icon } from "../components/atoms/index";
import { Error, Graphic } from "../components/molecules/index";
//Styles
import Colors from "../styles/Colors";
//API
import { USER_QUERY } from "../api/Queries";
import {
  USER_CONFIRM_TOKEN_MUTATION,
  USER_RESEND_CONFIRMATION_MUTATION
} from "../api/Mutations";
//Utils
import { getUrlParam, _refreshPage } from "../services/utilities";
import { ALERT_OPTIONS } from "../services/Constants";
import LogRocket from "logrocket";

export class ConfirmUser extends React.PureComponent {
  msg: () => mixed;
  props: {
    userQuery: any,
    confirmEmail: any,
    resendConfirmation: any,
    history: any,
    router: any
  };
  state = {
    confirmToken: "",
    error: false,
    errorMsg: ""
  };
  componentDidMount() {
    if (getUrlParam("token")) {
      this.setState({
        confirmToken: getUrlParam("token")
      });
    }
  }
  showAlert(
    type: string = "success",
    text: string = "Some Text",
    color: string = Colors.green,
    icon: string = "fa-link"
  ) {
    this.msg.show(text, {
      time: 3000,
      type,
      icon: <Icon name={icon} color={color} />
    });
  }
  _onSendConfirmationCode = async () => {
    //verifies the user and saves in the DB
    try {
      const { confirmToken } = this.state;
      await this.props.confirmEmail({
        variables: {
          confirmToken
        }
      });
      //redirects the user to the main page
      this.props.history.push("/");
    } catch (e) {
      LogRocket.error({ SendConfirmationCode: e });
      this.setState({
        error: true,
        errorMsg: "The verification code is invalid."
      });
    }
  };
  _onResendConfirmationCode = async () => {
    //If the code is invalid resend a new code
    try {
      const email = this.props.userQuery.user.email;
      await this.props.resendConfirmation({
        variables: {
          email
        }
      });
      this.showAlert(
        "success",
        "Your code has been resent, check your email.",
        Colors.green,
        "fa-envelope"
      );
    } catch (e) {
      LogRocket.error({ ResendConfirmationCode: e });
      this.setState({
        error: true,
        errorMsg: "The verification code is invalid."
      });
    }
  };
  render() {
    if (this.props.userQuery && this.props.userQuery.loading) {
      return <div>Loading</div>;
    }
    if (this.props.userQuery && this.props.userQuery.error) {
      return (
        <Graphic text="Ups! Something went wrong try again." icon="fa-plug">
          <Button
            className="btn btn-lg btn-primary"
            color={Colors.primary}
            onClick={_refreshPage}
          >
            Try Again
          </Button>
        </Graphic>
      );
    }
    if (this.props.userQuery.user === null) {
      this.props.history.push("/signin");
      return true;
    }
    if (this.props.userQuery.user.confirmed) {
      this.props.history.push("/");
      return true;
    }
    const { confirmToken, error, errorMsg } = this.state;
    return (
      <AuthLayout
        title="Please confirm your email"
        description="We like real people, we need to know if it's not a ghost of the internet."
      >
        <AlertContainer ref={(a) => (this.msg = a)} {...ALERT_OPTIONS} />
        <label htmlFor="confirmToken" className="sr-only">
          Confirmation Code
        </label>
        <Input
          id="confirmToken"
          value={confirmToken}
          onChange={(e) => this.setState({ confirmToken: e.target.value })}
          className="form-control"
          placeholder="Confirmation Code"
          required
          autoFocus
        />

        <Button
          className="btn btn-lg btn-primary btn-block"
          style={{ marginTop: 10, marginBottom: 10 }}
          color={Colors.primary}
          onClick={this._onSendConfirmationCode}
        >
          Confirm Account
        </Button>
        <Link onClick={this._onResendConfirmationCode}>
          Resend confirmation code?
        </Link>
        <Error show={error}>{errorMsg}</Error>
      </AuthLayout>
    );
  }
}

const confirmUserWithData = compose(
  graphql(USER_CONFIRM_TOKEN_MUTATION, { name: "confirmEmail" }),
  graphql(USER_RESEND_CONFIRMATION_MUTATION, { name: "resendConfirmation" }),
  graphql(USER_QUERY, { name: "userQuery" })
);

export default confirmUserWithData(ConfirmUser);
