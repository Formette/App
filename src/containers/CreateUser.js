// @flow
import React from "react";
import { graphql, compose, withApollo } from "react-apollo";
//Components
import AuthLayout from "../components/organisms/AuthLayout";
import { Input, Button, Link } from "../components/atoms/index";
import Error from "../components/molecules/Error";
//Styles
import Colors from "../styles/Colors";
//API
import { USERNAME_VALIDATION_QUERY } from "../api/Queries";
import { SIGIN_USER_MUTATION, CREATE_USER_MUTATION } from "../api/Mutations";
import { userSignIn } from "../api/Functions";
//Utils
import {
  generateToken,
  generateExpiration,
  _formatUsername
} from "../services/utilities";
import LogRocket from "logrocket";

export class CreateUser extends React.PureComponent {
  props: {
    createUser: any,
    signinUser: any,
    history: any,
    client: any,
    router: any
  };
  state = {
    email: "",
    password: "",
    username: "",
    error: false,
    errorMsg: "",
    timeoutUserName: 0,
    timeoutPassword: 0
  };
  _onCreateUser = () => {
    const { email, password, error } = this.state;
    let username = _formatUsername(this.state.username);
    //Verifies if the inputs are empty or not
    if (email && password && username) {
      if (error) {
        return;
      }
      if (this._checkPassword(password)) return;
      //Creates a new user
      this.props
        .createUser({
          variables: {
            email,
            password,
            username,
            confirmToken: generateToken(),
            confirmExpires: generateExpiration()
          }
        })
        .then(() => {
          LogRocket.track("Registered");
          this._onSignIn(email, password);
        })
        .catch(e => {
          LogRocket.error({ CreateUser: e });
          this.setState({
            error: true,
            errorMsg:
              "Ops! User already exists with that information, try again."
          });
        });
    } else {
      this.setState({
        error: true,
        errorMsg: "This form is feeling lonely, needs affection, needs data."
      });
    }
  };
  _onSignIn = async (email: string, password: string) => {
    //logs in the user
    const response = await userSignIn(email, password, this.props.signinUser);
    if (response.status) {
      LogRocket.track("Sign in");
      //checks if the user has the email confirmed
      if (response.confirmed) {
        //sends to the dashboard
        this.props.history.push("/");
      } else {
        this.props.history.push("/confirm");
      }
    } else {
      LogRocket.error({ SignIn: response });
      this.setState({
        error: true,
        errorMsg: "Ops! Something went wrong, try again."
      });
    }
  };
  _onPasswordValidation(password: string) {
    clearTimeout(this.state.timeoutPassword);
    this.setState({
      timeoutPassword: setTimeout(() => {
        this._checkPassword(password);
      }, 500)
    });
  }
  _checkPassword(password: string) {
    if (password.length <= 8) {
      LogRocket.warn(
        "With so much room in the box, you chose this tiny thing. We need more than 8 characters, go we know you can."
      );
      this.setState({
        error: true,
        errorMsg:
          "With so much room in the box, you chose this tiny thing. We need more than 8 characters, go we know you can."
      });
      return true;
    }
    this.setState({ error: false });
  }
  _onUsernameValidation(getUsername: string) {
    clearTimeout(this.state.timeoutUserName);
    this.setState({
      timeoutUserName: setTimeout(() => {
        let username = _formatUsername(getUsername);
        this.props.client
          .query({
            query: USERNAME_VALIDATION_QUERY,
            variables: { username }
          })
          .then(res => {
            if (Object.keys(res.data.allUsers).length !== 0) {
              LogRocket.info(
                "With so much name in this world, you had to choose this one. Try another."
              );
              this.setState({
                error: true,
                errorMsg:
                  "With so much name in this world, you had to choose this one. Try another."
              });
            } else {
              this.setState({ error: false });
            }
          })
          .catch(e => {
            LogRocket.error({ _onUsernameValidation: e });
          });
      }, 500)
    });
  }
  render() {
    const { email, password, username, error, errorMsg } = this.state;
    const { history } = this.props;
    return (
      <AuthLayout description="Seriously? Your forms do not have a home for the data? Do not worry, they will not be without shelter.">
        <label htmlFor="signupUsername" className="sr-only">
          Username
        </label>
        <Input
          id="signupUsername"
          value={username}
          onChange={e => this.setState({ username: e.target.value })}
          onKeyUp={e => this._onUsernameValidation(e.target.value)}
          className="form-control"
          placeholder="Username"
          required
          autoFocus
        />
        <label htmlFor="signupEmail" className="sr-only">
          Email address
        </label>
        <Input
          id="signupEmail"
          type="email"
          value={email}
          onChange={e => this.setState({ email: e.target.value })}
          className="form-control"
          placeholder="Email address"
          required
          autoFocus
        />

        <label htmlFor="signupPassword" className="sr-only">
          Password
        </label>
        <Input
          id="signupPassword"
          type="password"
          value={password}
          onChange={e => this.setState({ password: e.target.value })}
          onKeyUp={e => this._onPasswordValidation(e.target.value)}
          className="form-control"
          placeholder="Password"
          required
          autoFocus
        />

        <Button
          className="btn btn-lg btn-primary btn-block"
          onClick={this._onCreateUser}
          style={{ marginTop: 10, marginBottom: 10 }}
          color={Colors.primary}
        >
          Create Account
        </Button>

        <Link onClick={() => history.push("/signin")}>
          Already have an account? Your forms miss you.{" "}
          <u>
            <strong>Sign In</strong>
          </u>
        </Link>
        <Error show={error}>{errorMsg}</Error>
      </AuthLayout>
    );
  }
}

const CreateUserWithData = compose(
  graphql(CREATE_USER_MUTATION, { name: "createUser" }),
  graphql(SIGIN_USER_MUTATION, { name: "signinUser" })
);

export default withApollo(CreateUserWithData(CreateUser));
