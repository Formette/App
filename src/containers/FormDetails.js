// @flow
import React, {PureComponent} from 'react';
import { graphql, compose } from "react-apollo";
//Components
import CopyToClipboard from 'react-copy-to-clipboard';
import AlertContainer from 'react-alert';
import {SubTitle, Header, Button, Icon} from '../components/atoms/index';
import {HorizontalList, Table, Graphic} from '../components/molecules/index';
//Styles
import Colors from '../styles/Colors';
//Utils
import {ALERT_OPTIONS} from '../services/Constants';
import { _getUsername, _refreshPage, _getUserId} from "../services/utilities";
//API
import { FORM_DATA_QUERY, ALL_FORMS_QUERY } from "../api/Queries";
import { DELETE_FORM_MUTATION } from "../api/Mutations";
import { FORM_DATA_SUBSCRIPTION } from "../api/Subscriptions";

const url = `api.formette.com/${_getUsername()}/`;

class FormDetails extends PureComponent{
    msg: any;
    props: {
        formDataQuery: any,
        deleteFormMutation: any,
    };
    componentWillMount() {
        this._subscribeToNewLinks();
    }
    _subscribeToNewLinks = () => {
        this.props.formDataQuery.subscribeToMore({
            document: FORM_DATA_SUBSCRIPTION,
            variables: { id: this.props.match.params.id },
            updateQuery: (previous, { subscriptionData }) => {
                console.log("subscriptionData = ", subscriptionData);
                console.log("previous = ", previous);
                const newItems = [
                    subscriptionData.data.Forms.node.data,
                    ...previous.Forms.data
                ];
                const result = {
                    ...previous,
                    data: newItems
                };
                return result
            }
        });
    };
    showAlert(type: string = "success", text: string = "Some Text", color: string = Colors.green, icon: string = "fa-link"){
        this.msg.show(text, {
            time: 3000,
            type: type,
            icon: <Icon name={icon} color={color}/>
        })
    }
    _deleteForm = async () => {
        //deletes the form in the DB
        try{
            const {id} = this.props.match.params;
            const userId = _getUserId();
            await this.props.deleteFormMutation({
                variables: {
                    id,
                },
                update: (store, { data: {deleteForms} }) => {
                    try {
                        //reads the query from the cache
                        const data = store.readQuery({ query: ALL_FORMS_QUERY, variables: {userId: userId} });
                        //finds and removes the form from the object
                        data.allFormses.forEach((value, index) => {
                            if (value.id === deleteForms.id) {
                                data.allFormses.splice(index, 1);
                            }
                        });
                        //updates the new data to the store
                        store.writeQuery({ query: ALL_FORMS_QUERY, variables: {userId: userId}, data });
                    } catch (e) {
                        console.error(e);
                    }
                }
            });
            //Shows feedback and updates the store
            //this.showAlert("success", "Form created successfully");
            this.props.history.push("/");
        }catch(e){
            console.error(e);
        }
    };
    _editForm = _ => {
        this.props.history.push(`/edit/${this.props.match.params.id}`)
    };
    render(){
        console.log("data = ",this.props.formDataQuery);
        if (this.props.formDataQuery && this.props.formDataQuery.loading) {
            return <div>Loading</div>;
        }
        if (this.props.formDataQuery && this.props.formDataQuery.error) {
            return <Graphic text="Ups! Something went wrong try again." icon="fa-plug">
                        <Button className="btn btn-lg btn-primary"
                                color={Colors.primary}
                                onClick={_refreshPage}>
                            Try Again
                        </Button>
                   </Graphic>
        }
        if (this.props.formDataQuery.Forms === null) {
            return <Graphic
                        text="Hello Indiana Jones, are you in unfamiliar lands? You are a great explorer but this form does
                              not exist and may contain mysterious dangers. Come back home."
                        >
                    <Button className="btn btn-lg btn-primary"
                            color={Colors.primary}
                            onClick={_ => this.props.history.push('/')}>
                        Go back home
                    </Button>
                </Graphic>
        }
        const {data: items, name, endpoint} = this.props.formDataQuery.Forms;
        const point = endpoint.split("/");
       // const point = "teste";
        return(
            <div>
                <AlertContainer ref={a => this.msg = a} {...ALERT_OPTIONS} />
                <div className="row">
                    <div className="col-md-6 col-sm-6">
                        <SubTitle text="All the data for" color={Colors.text.secondary}/>
                        <Header className="text-truncate" text={name}/>
                    </div>
                    <div className="col-md-6 col-sm-6">
                        <HorizontalList className="float-right">
                            <li>
                                <CopyToClipboard text={`${url}${point[1]}`}
                                                 style={{cursor: "pointer"}}
                                                 onCopy={_ => this.showAlert("success", "Endpoint copied to clipboard")}>
                                    <Button className="btn" color={Colors.default}>
                                        <Icon color={Colors.white} name="fa-link"/>
                                    </Button>
                                </CopyToClipboard>
                            </li>
                            <li>
                                <Button className="btn" color={Colors.default} onClick={this._editForm}>
                                    <Icon color={Colors.white} name="fa-pencil"/>
                                </Button>
                            </li>
                            <li>
                                <Button className="btn" color={Colors.red} onClick={this._deleteForm}>
                                    <Icon color={Colors.white} name="fa-trash-o"/>
                                </Button>
                            </li>
                        </HorizontalList>
                    </div>
                </div>
                <div className="row">
                    <div className="col-md-12">
                        <div className="card">
                            <div className="card-body">
                                {Object.keys(items).length === 0 ?
                                    <div>This form is so sad ... You do not have any submissions yet, help it, go! Look for data for this poor guy.</div>
                                    :
                                    <Table data={items}/>
                                }
                            </div>
                        </div>

                    </div>
                </div>
            </div>
        )
    }
}

const FormDetailsWithData = compose(
    graphql(FORM_DATA_QUERY, { name: "formDataQuery",
        options: (props) => ({
            variables: {id: props.match.params.id}
        })
    }),
    graphql(DELETE_FORM_MUTATION, { name: "deleteFormMutation" })
)(FormDetails);

export default FormDetailsWithData;