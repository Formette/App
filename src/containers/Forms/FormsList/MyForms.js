import React, { Component, Fragment } from "react";
import PropTypes from "prop-types";
import { graphql, compose } from "react-apollo";
import { withRouter } from "react-router";
//Container
import Tools from "./Tools";
import Cards from "./Cards";
//Components
import { Button, Icon, InputGroup } from "../../../components/atoms/index";
import { Graphic, Loader } from "../../../components/molecules/index";
//Utils
import { _refreshPage, _onHandleExpression } from "../../../services/utilities";
//API
import { ALL_FORMS_QUERY } from "../../../api/Queries";
//locales
import { FormattedMessage, injectIntl } from "react-intl";
import { globals as messages } from "../../../locales/api";

export class MyForms extends Component {
  static propTypes = {
    allFormsQuery: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired,
    alert: PropTypes.object,
    intl: PropTypes.object.isRequired
  };
  state = {
    isLoading: true,
    isSearching: false,
    timeoutSearchFilter: 0,
    searchQuery: "",
    data: [],
    dataStatic: []
  };
  componentWillReceiveProps(nextProps) {
    const { allFormses } = nextProps.allFormsQuery;
    if (!nextProps.allFormsQuery.loading) {
      if (Object(allFormses).keys) {
        this.setState({
          data: allFormses,
          dataStatic: allFormses,
          isLoading: false
        });
        return;
      }
    }
  }
  _goToNew = () => {
    this.props.history.push("/new");
  };
  _onSearchFilter() {
    clearTimeout(this.state.timeoutSearchFilter);
    this.setState({
      isSearching: true,
      timeoutSearchFilter: setTimeout(() => {
        const { searchQuery, dataStatic } = this.state;
        const { allFormses } = this.props.allFormsQuery;
        try {
          let search = new RegExp(
            _onHandleExpression(searchQuery.toLowerCase().trim())
          );
          //if the textbox is empty show all the dashboards again
          if (!searchQuery) {
            this.setState({ data: allFormses, isSearching: false });
            return;
          }
          //Filters dashboard with keyword on the search box
          let filtered = dataStatic.filter(item => {
            return (
              item.name
                .toLowerCase()
                .trim()
                .search(search) !== -1
            );
          });
          //Sets and shows the result
          this.setState({
            data: filtered,
            isSearching: false
          });
        } catch (error) {
          this.props.alert.error("error on search");
        }
      }, 200)
    });
  }
  render() {
    const { allFormsQuery, intl } = this.props;
    const { allFormses } = allFormsQuery;
    const { isLoading, isSearching, data } = this.state;
    if (this.props.allFormsQuery.error) {
      return (
        <Graphic
          title={intl.formatMessage(messages.GraphicErrorTitle)}
          description={intl.formatMessage(messages.GraphicErrorDescription)}
          imgType="error"
          top={200}
        >
          <Button
            className="btn btn-lg btn-primary"
            onClick={_refreshPage}
            primary
          >
            <FormattedMessage
              id="app.graphic.error.action"
              defaultMessage={"Try Again"}
            />
          </Button>
        </Graphic>
      );
    }
    return (
      <Fragment>
        <Tools
          title={intl.formatMessage(messages.PageFormTitle)}
          description={intl.formatMessage(messages.PageFormDescription)}
          titleTruncate
          textTruncate
        >
          <div className="col">
            <InputGroup
              InputProps={{
                type: "text",
                className: "form-control",
                placeholder: intl.formatMessage(
                  messages.PageFormSearchPlaceholder
                ),
                defaultValue: this.state.searchQuery
              }}
              IconProps={{ name: "fas fa-search" }}
              onChange={e => this.setState({ searchQuery: e.target.value })}
              onKeyUp={e => this._onSearchFilter(e.target.value)}
            />
          </div>
          <div className="col">
            <Button className="btn btn-lg" onClick={this._goToNew} primary>
              <Icon name="fas fa-plus" color="#FFF" />{" "}
              <FormattedMessage
                id="app.page.form.action.new.form"
                defaultMessage={"New form"}
              />
            </Button>
          </div>
        </Tools>
        <div className="row">
          {isLoading || isSearching ? (
            <Loader top={100} />
          ) : Object.keys(allFormses).length === 0 ? (
            <Graphic
              title={intl.formatMessage(messages.GraphicFormEmptyTitle)}
              description={intl.formatMessage(
                messages.GraphicFormEmptyDescription
              )}
              imgType="empty"
              top={55}
            />
          ) : (
            <Cards data={data} />
          )}
        </div>
      </Fragment>
    );
  }
}

export default compose(
  injectIntl,
  withRouter,
  graphql(ALL_FORMS_QUERY, {
    name: "allFormsQuery",
    options: props => ({
      variables: { userId: props.userId }
    })
  })
)(MyForms);
