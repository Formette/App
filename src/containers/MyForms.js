// @flow
import React, { Component } from "react";
import { graphql, compose } from "react-apollo";
import * as moment from "moment";
//Components
import { Header, Button } from "../components/atoms/index";
import {
  Placeholder,
  PlaceholderAnimation,
  Card,
  Graphic
} from "../components/molecules/index";
//Utils
import { _refreshPage, _getUserId } from "../services/utilities";
import LogRocket from "logrocket";
//Styles
import Colors from "../styles/Colors";
//API
import { ALL_FORMS_QUERY } from "../api/Queries";

export class MyForms extends Component {
  props: {
    allFormsQuery: any,
    history: any
  };
  state = {
    loading: true,
    loadingTimeout: 0,
    hasError: "",
    data: []
  };
  _LoadingAnimationContent = (type: string = "loading", length: number = 6) => {
    const content = [];
    for (let i = 0; i <= length; i++) {
      if (type !== "loading") {
        content.push(<Placeholder key={i} width={208} height={193} />);
      } else {
        content.push(<PlaceholderAnimation key={i} />);
      }
    }
    return content;
  };
  _loadContent = () => {
    const { allFormses } = this.props.allFormsQuery;
    const formsCount = Object.keys(allFormses).length;
    const content = [];
    let length = 0;
    //checks if the number of forms created from the user
    if (formsCount === 0) {
      return this._LoadingAnimationContent("normal");
    }
    //Checks if the object data is not empty
    if (formsCount >= 0) {
      allFormses.map((res) => {
        content.push(
          <Card
            key={res.id}
            title={res.name}
            date={moment(res.createdAt).format("ll")}
            onClick={() => this.props.history.push(`/form/${res.id}`)}
          />
        );
        return true;
      });
      //for interface
      if (formsCount < 6) {
        length = 6 - Number(formsCount);
      }
      LogRocket.track("Loaded Forms");
      return content.concat(this._LoadingAnimationContent("normal", length));
    }
  };
  _goToNew = () => {
    this.props.history.push("/new");
  };
  render() {
    if (this.props.allFormsQuery && this.props.allFormsQuery.error) {
      return (
        <Graphic text="Ups! Something went wrong try again." icon="fa-plug">
          <Button
            className="btn btn-lg btn-primary"
            color={Colors.primary}
            onClick={_refreshPage}
          >
            Try Again
          </Button>
        </Graphic>
      );
    }
    return (
      <div className="row">
        <div className="col-md-12 col-sm-12 col">
          <Header text="My Forms" />
          <Card
            icon="fa-plus"
            title="New Form"
            date="Create from scratch"
            onClick={this._goToNew}
            new
          />
          {this.props.allFormsQuery.loading ? (
            this._LoadingAnimationContent()
          ) : (
            <div>{this._loadContent()}</div>
          )}
        </div>
      </div>
    );
  }
}

const myFormsWithData = compose(
  graphql(ALL_FORMS_QUERY, {
    name: "allFormsQuery",
    options: (props) => ({
      variables: { userId: _getUserId() }
    })
  })
)(MyForms);

export default myFormsWithData;
