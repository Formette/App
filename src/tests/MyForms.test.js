// MyForms.test.js
//Utils
import './helpers/browser';
import './helpers/mock-localstorage';
import React from 'react';
import {MyForms} from '../containers/MyForms';
import renderer from 'react-test-renderer';

describe('My Forms', () => {
    it('UI renders correctly', () => {
        const data = {
            loading: true,
        };
        const tree = renderer.create(
            <MyForms allFormsQuery={data}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

