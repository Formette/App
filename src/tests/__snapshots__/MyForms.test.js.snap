// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Forms UI renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-md-12 col-sm-12 col"
  >
    <h2
      className="sc-bxivhb kRbHZm"
    >
      My Forms
    </h2>
    <div
      className="card text-center sc-gZMcBi iWAETB"
      onClick={[Function]}
    >
      <div
        className="card-body"
      >
        <i
          className="fa fa-plus sc-gzVnrw hIWOOg"
          style={undefined}
        >
          ‌
        </i>
        <h3
          className="card-title text-truncate sc-ifAKCX bljzpw"
        >
          New Form
        </h3>
        <h6
          className="card-subtitle mb-2  sc-bZQynM dIEjhl"
        >
          Create from scratch
        </h6>
      </div>
    </div>
    <div
      className="card sc-jTzLTM dlGUEl sc-VigVT istVuo"
    >
      <div
        className="card-body"
      >
        <div
          className="animated-background"
        >
          <div
            className="background-masker header-top"
          >
            ‌
          </div>
          <div
            className="background-masker header-left"
          >
            ‌
          </div>
          <div
            className="background-masker header-right"
          >
            ‌
          </div>
          <div
            className="background-masker header-bottom"
          >
            ‌
          </div>
          <div
            className="background-masker subheader-left"
          >
            ‌
          </div>
          <div
            className="background-masker subheader-right"
          >
            ‌
          </div>
          <div
            className="background-masker subheader-bottom"
          >
            ‌
          </div>
          <div
            className="background-masker content-top"
          >
            ‌
          </div>
          <div
            className="background-masker content-first-end"
          >
            ‌
          </div>
          <div
            className="background-masker content-second-line"
          >
            ‌
          </div>
          <div
            className="background-masker content-second-end"
          >
            ‌
          </div>
          <div
            className="background-masker content-third-line"
          >
            ‌
          </div>
          <div
            className="background-masker content-third-end"
          >
            ‌
          </div>
        </div>
      </div>
    </div>
    <div
      className="card sc-jTzLTM dlGUEl sc-VigVT istVuo"
    >
      <div
        className="card-body"
      >
        <div
          className="animated-background"
        >
          <div
            className="background-masker header-top"
          >
            ‌
          </div>
          <div
            className="background-masker header-left"
          >
            ‌
          </div>
          <div
            className="background-masker header-right"
          >
            ‌
          </div>
          <div
            className="background-masker header-bottom"
          >
            ‌
          </div>
          <div
            className="background-masker subheader-left"
          >
            ‌
          </div>
          <div
            className="background-masker subheader-right"
          >
            ‌
          </div>
          <div
            className="background-masker subheader-bottom"
          >
            ‌
          </div>
          <div
            className="background-masker content-top"
          >
            ‌
          </div>
          <div
            className="background-masker content-first-end"
          >
            ‌
          </div>
          <div
            className="background-masker content-second-line"
          >
            ‌
          </div>
          <div
            className="background-masker content-second-end"
          >
            ‌
          </div>
          <div
            className="background-masker content-third-line"
          >
            ‌
          </div>
          <div
            className="background-masker content-third-end"
          >
            ‌
          </div>
        </div>
      </div>
    </div>
    <div
      className="card sc-jTzLTM dlGUEl sc-VigVT istVuo"
    >
      <div
        className="card-body"
      >
        <div
          className="animated-background"
        >
          <div
            className="background-masker header-top"
          >
            ‌
          </div>
          <div
            className="background-masker header-left"
          >
            ‌
          </div>
          <div
            className="background-masker header-right"
          >
            ‌
          </div>
          <div
            className="background-masker header-bottom"
          >
            ‌
          </div>
          <div
            className="background-masker subheader-left"
          >
            ‌
          </div>
          <div
            className="background-masker subheader-right"
          >
            ‌
          </div>
          <div
            className="background-masker subheader-bottom"
          >
            ‌
          </div>
          <div
            className="background-masker content-top"
          >
            ‌
          </div>
          <div
            className="background-masker content-first-end"
          >
            ‌
          </div>
          <div
            className="background-masker content-second-line"
          >
            ‌
          </div>
          <div
            className="background-masker content-second-end"
          >
            ‌
          </div>
          <div
            className="background-masker content-third-line"
          >
            ‌
          </div>
          <div
            className="background-masker content-third-end"
          >
            ‌
          </div>
        </div>
      </div>
    </div>
    <div
      className="card sc-jTzLTM dlGUEl sc-VigVT istVuo"
    >
      <div
        className="card-body"
      >
        <div
          className="animated-background"
        >
          <div
            className="background-masker header-top"
          >
            ‌
          </div>
          <div
            className="background-masker header-left"
          >
            ‌
          </div>
          <div
            className="background-masker header-right"
          >
            ‌
          </div>
          <div
            className="background-masker header-bottom"
          >
            ‌
          </div>
          <div
            className="background-masker subheader-left"
          >
            ‌
          </div>
          <div
            className="background-masker subheader-right"
          >
            ‌
          </div>
          <div
            className="background-masker subheader-bottom"
          >
            ‌
          </div>
          <div
            className="background-masker content-top"
          >
            ‌
          </div>
          <div
            className="background-masker content-first-end"
          >
            ‌
          </div>
          <div
            className="background-masker content-second-line"
          >
            ‌
          </div>
          <div
            className="background-masker content-second-end"
          >
            ‌
          </div>
          <div
            className="background-masker content-third-line"
          >
            ‌
          </div>
          <div
            className="background-masker content-third-end"
          >
            ‌
          </div>
        </div>
      </div>
    </div>
    <div
      className="card sc-jTzLTM dlGUEl sc-VigVT istVuo"
    >
      <div
        className="card-body"
      >
        <div
          className="animated-background"
        >
          <div
            className="background-masker header-top"
          >
            ‌
          </div>
          <div
            className="background-masker header-left"
          >
            ‌
          </div>
          <div
            className="background-masker header-right"
          >
            ‌
          </div>
          <div
            className="background-masker header-bottom"
          >
            ‌
          </div>
          <div
            className="background-masker subheader-left"
          >
            ‌
          </div>
          <div
            className="background-masker subheader-right"
          >
            ‌
          </div>
          <div
            className="background-masker subheader-bottom"
          >
            ‌
          </div>
          <div
            className="background-masker content-top"
          >
            ‌
          </div>
          <div
            className="background-masker content-first-end"
          >
            ‌
          </div>
          <div
            className="background-masker content-second-line"
          >
            ‌
          </div>
          <div
            className="background-masker content-second-end"
          >
            ‌
          </div>
          <div
            className="background-masker content-third-line"
          >
            ‌
          </div>
          <div
            className="background-masker content-third-end"
          >
            ‌
          </div>
        </div>
      </div>
    </div>
    <div
      className="card sc-jTzLTM dlGUEl sc-VigVT istVuo"
    >
      <div
        className="card-body"
      >
        <div
          className="animated-background"
        >
          <div
            className="background-masker header-top"
          >
            ‌
          </div>
          <div
            className="background-masker header-left"
          >
            ‌
          </div>
          <div
            className="background-masker header-right"
          >
            ‌
          </div>
          <div
            className="background-masker header-bottom"
          >
            ‌
          </div>
          <div
            className="background-masker subheader-left"
          >
            ‌
          </div>
          <div
            className="background-masker subheader-right"
          >
            ‌
          </div>
          <div
            className="background-masker subheader-bottom"
          >
            ‌
          </div>
          <div
            className="background-masker content-top"
          >
            ‌
          </div>
          <div
            className="background-masker content-first-end"
          >
            ‌
          </div>
          <div
            className="background-masker content-second-line"
          >
            ‌
          </div>
          <div
            className="background-masker content-second-end"
          >
            ‌
          </div>
          <div
            className="background-masker content-third-line"
          >
            ‌
          </div>
          <div
            className="background-masker content-third-end"
          >
            ‌
          </div>
        </div>
      </div>
    </div>
    <div
      className="card sc-jTzLTM dlGUEl sc-VigVT istVuo"
    >
      <div
        className="card-body"
      >
        <div
          className="animated-background"
        >
          <div
            className="background-masker header-top"
          >
            ‌
          </div>
          <div
            className="background-masker header-left"
          >
            ‌
          </div>
          <div
            className="background-masker header-right"
          >
            ‌
          </div>
          <div
            className="background-masker header-bottom"
          >
            ‌
          </div>
          <div
            className="background-masker subheader-left"
          >
            ‌
          </div>
          <div
            className="background-masker subheader-right"
          >
            ‌
          </div>
          <div
            className="background-masker subheader-bottom"
          >
            ‌
          </div>
          <div
            className="background-masker content-top"
          >
            ‌
          </div>
          <div
            className="background-masker content-first-end"
          >
            ‌
          </div>
          <div
            className="background-masker content-second-line"
          >
            ‌
          </div>
          <div
            className="background-masker content-second-end"
          >
            ‌
          </div>
          <div
            className="background-masker content-third-line"
          >
            ‌
          </div>
          <div
            className="background-masker content-third-end"
          >
            ‌
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
